src/mito2/src/manifest/manager.rs

```diff
@@ -472,6 +478,28 @@ impl RegionManifestManagerInner {
             Ok(None)
         }
     }
+
+    /// Returns true if a newer version manifest file is found.
+    ///
+    /// It is typically used in read-only regions to catch up with manifest.
+    pub(crate) async fn check_update(&self) -> Result<bool> {
+        let last_version = self.last_version;
+        let newer_manifests = self
+            .store
+            .get_paths(|entry| {
+                let file_name = entry.name();
+                if is_delta_file(file_name) {
+                    let version = file_version(file_name);
+                    if version > last_version {
+                        return Some((version, entry));
+                    }
+                }
+                None
+            })
+            .await?;
+
+        Ok(!newer_manifests.is_empty())
```

Comment:
We don't need the details of manifests files. Since there's no pub iterator on manifest files, can we rewrite the codes like this, saving some regex match costs:

```suggestion
        let mut found = false;
        
        let _ = self
            .store
            .get_paths(|entry| {
                if !found {
                    let file_name = entry.name();
                    if is_delta_file(file_name) {
                        let version = file_version(file_name);
                        if version > last_version {
                            found = true;
                        }
                    }
                }
                None
            })
            .await?;

        Ok(found)
```

Reply:
It could be tricky; there is an ownership issue. BTW, maybe we can refactor `is_delta_file` to use `right_match`  instead of a regex match.

Reply:
I decided to use `try_any`